### ISSUE TITLE
UserWarning: linear_trend_timewise requires the data to have a index of type <class 'pandas.core.indexes.datetimes.DatetimeIndex'>. Results will not be calculated

### ISSUE THREAD

**Issue by forhonourlx** [blue-yonder/tsfresh#557]
Hi,

I am suffering this warning, could somebody know how to solve this problem?
/home/simon/anaconda3/lib/python3.6/site-packages/tsfresh/feature_extraction/extraction.py:380: UserWarning: linear_trend_timewise requires the data to have a index of type <class 'pandas.core.indexes.datetimes.DatetimeIndex'>. Results will not be calculated

Thanks.

**Comment by advahad** [comment]
Having the same issue, any luck?

**Comment by xenmind** [comment]
Pretty sure this is explained here: https://github.com/blue-yonder/tsfresh/blob/master/notebooks/feature_extraction_with_Datetime_index.ipynb A feature set needs date/time index which you aren't providing.

**Comment by forhonourlx** [comment]
@xenmind Thank you for your help.
I am following this example [https://github.com/blue-yonder/tsfresh/blob/master/notebooks/timeseries_forecasting_google_stock.ipynb](url)

Codes:
`ser_singular = pd.Series(data_np[i_,:,ii_], index=self.date_pd)
                print('instrument:',i_,',idx:',ii_,',ser_singular:\n',ser_singular)

                df_shift, y = make_forecasting_frame(ser_singular, kind="price", max_timeshift=tsfresh_rollback, rolling_direction=1)
                print('instrument:',i_,',idx:',ii_,',df_shift:\n',df_shift.head())
                #print('y:\n',y.head())
                X = tsfresh.extract_features(df_shift, column_id="id", column_sort="time", column_value="value", impute_function=impute, show_warnings=False)
                print('instrument:',i_,',idx:',ii_,',features:\n',X.head())`

Then got:
/home/simon/anaconda3/lib/python3.6/site-packages/tsfresh/feature_extraction/extraction.py:380: UserWarning: linear_trend_timewise requires the data to have a index of type <class 'pandas.core.indexes.datetimes.DatetimeIndex'>. Results will not be calculated
  "not be calculated".format(function_name, index_type)

Is there anything to deal when tsfresh.utilities.dataframe_functions.make_forecasting_frame() ？

**Comment by forhonourlx** [comment]
instrument: 0 ,idx: 5 ,ser_singular:
 Date
2013-09-15 21:16:00     0.000000
...
2013-09-20 20:57:00    48.338407
2013-09-20 20:58:00    64.610400
2013-09-20 20:59:00    62.185380
2013-09-20 21:00:00    51.245968
Length: 7125, dtype: float64
/home/simon/anaconda3/lib/python3.6/site-packages/tsfresh/utilities/dataframe_functions.py:439: UserWarning: Your time stamps are not uniformly sampled, which makes rolling nonsensical in some domains.
  warnings.warn("Your time stamps are not uniformly sampled, which makes rolling "
instrument: 0 ,idx: 5 ,df_shift:
                       time  value                  id   kind
896620 2013-09-15 21:16:00    0.0 2013-09-15 21:17:00  price
889497 2013-09-15 21:16:00    0.0 2013-09-15 21:18:00  price
896621 2013-09-15 21:17:00    0.0 2013-09-15 21:18:00  price
882375 2013-09-15 21:16:00    0.0 2013-09-15 21:19:00  price
889498 2013-09-15 21:17:00    0.0 2013-09-15 21:19:00  price

**Comment by forhonourlx** [comment]
`df_shift.index = pd.DatetimeIndex(df_shift['time'])`
Is it OK?

**Comment by andrewlarimer** [comment]
I am experiencing this same issue. Here is a snippet of the code:

```
x_train_imputed.index = pd.DatetimeIndex(x_train_imputed.agg_timestamp)
x_train_imputed.agg_timestamp = pd.DatetimeIndex(x_train_imputed.agg_timestamp)

extracted_features = extract_features(x_train_imputed.drop(columns=['Site', 'UnitId', 'target']),
                      column_id='example_id', column_sort="agg_timestamp", distributor=Distributor)
```

So I am converting both the column_sort column as well as the index to the pd.DatetimeIndex type, but I still get the same warning for seemingly all features:

UserWarning: linear_trend_timewise requires the data to have a index of type <class 'pandas.core.indexes.datetimes.DatetimeIndex'>. Results will not be calculated

**Comment by nils-braun** [comment]
There was a fix in the PR #576. I assume it fixed the issue. If not, please reopen :-)
